cmd/milvus/mck.go

```diff
@@ -0,0 +1,692 @@
+		line()
+		fmt.Println("Invalid Paths")
+		for id, paths := range c.taskIDToInvalidPath {
+			fmt.Printf("Task ID: %d\n", id)
+			for _, path := range paths {
+				fmt.Printf("\t%s\n", path)
+			}
+			invalidTasksOfPath = append(invalidTasksOfPath, id)
+		}
+		invalidTasksOfPath = removeRepeatElement(invalidTasksOfPath)
+	}
+
+	// combine all invalid tasks
+	var invalidTasks []int64
+	invalidTasks = append(invalidTasks, invalidTasksOfPartition...)
+	invalidTasks = append(invalidTasks, invalidTasksOfSegment...)
+	invalidTasks = append(invalidTasks, invalidTasksOfPath...)
+	invalidTasks = removeRepeatElement(invalidTasks)
+	if len(invalidTasks) > 0 {
+		line()
+		fmt.Println("All invalid tasks")
+		for _, invalidTask := range invalidTasks {
+			line2()
+			fmt.Printf("TaskID: %d\t%s\n", invalidTask, c.taskNameMap[invalidTask])
+		}
+	}
+	c.handleUserEnter(invalidTasks)
+
+}
+
+func (c *mck) initParam() {
+	c.taskKeyMap = make(map[int64]string)
+	c.taskNameMap = make(map[int64]string)
+	c.allTaskInfo = make(map[string]string)
+	c.partitionIDToTasks = make(map[int64][]int64)
+	c.segmentIDToTasks = make(map[int64][]int64)
+	c.taskIDToInvalidPath = make(map[int64][]string)
+	c.segmentIDs = make(map[int64]*datapb.SegmentInfo)
+	c.partitionIDs = make(map[int64]interface{})
+}
+
+func (c *mck) formatFlags(args []string, flags *flag.FlagSet) {
+	flags.StringVar(&c.etcdIP, "etcdIp", "", "Etcd endpoint to connect")
+	flags.StringVar(&c.ectdRootPath, "etcdRootPath", "", "Etcd root path")
+	flags.StringVar(&c.minioAddress, "minioAddress", "", "Minio endpoint to connect")
+	flags.StringVar(&c.minioUsername, "minioUsername", "", "Minio username")
+	flags.StringVar(&c.minioPassword, "minioPassword", "", "Minio password")
+	flags.StringVar(&c.minioUseSSL, "minioUseSSL", "", "Minio to use ssl")
+	flags.StringVar(&c.minioBucketName, "minioBucketName", "", "Minio bucket name")
+
+	if err := flags.Parse(os.Args[2:]); err != nil {
+		log.Fatal("failed to parse flags", zap.Error(err))
+	}
+	log.Info("args", zap.Strings("args", args))
+}
+
+func (c *mck) connectEctd() {
+	c.params.Init()
+	var etcdCli *clientv3.Client
+	var err error
+	if c.etcdIP != "" {
+		etcdCli, err = etcd.GetRemoteEtcdClient([]string{c.etcdIP})
+	} else {
+		etcdCli, err = etcd.GetEtcdClient(&c.params.EtcdCfg)
+	}
+	if err != nil {
+		log.Fatal("failed to connect to etcd", zap.Error(err))
+	}
+
+	rootPath := getNotEmptyString(c.ectdRootPath, c.params.EtcdCfg.MetaRootPath)
+	c.etcdKV = etcdkv.NewEtcdKV(etcdCli, rootPath)
+	log.Info("Etcd root path", zap.String("root_path", rootPath))
+}
+
+func (c *mck) connectMinio() {
+	useSSL := c.params.MinioCfg.UseSSL
+	if c.minioUseSSL == "true" || c.minioUseSSL == "false" {
+		useSSL, _ = strconv.ParseBool(c.minioUseSSL)
+	}
+	chunkManagerFactory := storage.NewChunkManagerFactory("local", "minio",
+		storage.RootPath(c.params.LocalStorageCfg.Path),
+		storage.Address(getNotEmptyString(c.minioAddress, c.params.MinioCfg.Address)),
```

Comment:
firstly,  optional parameters have higher priority,  we should check parameters' integrity once use them.
then try to use default configurations from milvus.yaml